docs/guides/stretch.mdx

```diff
@@ -0,0 +1,162 @@
+
+![Image illustrating the following circuit](/docs/images/guides/stretch/stretch.avif)
+
+The corresponding circuit would look like the following. Note that a pair of barriers is needed to define the boundaries of this relative timing.
+
+```python
+from qiskit.circuit import QuantumCircuit, QuantumRegister, ClassicalRegister
+
+qubits = QuantumRegister(2)
+clbits = ClassicalRegister(2)
+circuit = QuantumCircuit(qubits, clbits)
+(q0, q1) = qubits
+(c0, c1) = clbits
+
+# Add barriers to define the boundaries
+circuit.barrier()
+circuit.h(q0)
+circuit.measure(q0, c0)
+with circuit.if_test((c0, 1)) as else_:
+    circuit.h(q0)
+with else_:
+    circuit.x(q0)
+
+# Apply an XX DD sequence with stretch on qubit 1
+s = circuit.add_stretch("s")
+circuit.delay(s, q1)
+circuit.x(q1)
+circuit.delay(expr.mul(s, 2), q1)
+circuit.x(q1)
+circuit.delay(s, q1)
+circuit.barrier()
+```
+
+Here is another example of using `stretch` to ensure a sequence of gates between two barriers are left-aligned, whatever their actual durations are:
+```python
+from qiskit import QuantumCircuit
+from numpy import pi
+
+qc = QuantumCircuit(5)
+qc.barrier()
+qc.cx(0, 1)
+qc.u(pi/4, 0, pi/2, 2)
+qc.cx(3, 4)
+
+a = qc.add_stretch("a")
+b = qc.add_stretch("b")
+c = qc.add_stretch("c")
+
+# Use the stretches as Delay duration.
+qc.delay(a, [0, 1])
+qc.delay(b, 2)
+qc.delay(c, [3, 4])
+qc.barrier()
+```
+
+## Limitations
+
+The following constraints apply when using `stretch`:
+
+- At most one stretch variable per _qubit set_ between barriers (implicit and explicit). A qubit set is one or more qubits; these sets must be mutually exclusive.
+    Valid:
+
+        ```python
+        stretch a;
+        stretch b;
+        delay[a] $0, $1;
+        delay[b] $2
+        ```
+
+    Invalid:
+
+        ```python
+        stretch a;
+        stretch b;
+        delay[a] $0, $1;
+        delay[b] $0;
+        ```
+
+- A stretch variable cannot be used between multiple sets of barriers.
+        Valid:
+
+        ```python
+        barrier $0, $1;
+        delay[a] $0;
+        barrier $3;
+        delay[a] $0;
+        barrier $0, $1;
+        ```
+
+    Invalid:
+
+        ```python
+        barrier $0, $1;
+        delay[a] $0;
+        barrier $0, $1;
+        delay[a] $0;
+        barrier $0, $1;
+         ```
+
+- Stretch expressions are limited to those of the form `A*stretch + B` where `A` and `B` are floating point or integer constants.
+        Valid:
+
+        ```python
+        delay[2 * a + 3.0] $0;
+        delay[a / 5 - 40.0] $0;
+        delay[a + a] $0; # equivalent to "delay[2 * a] $0;"
+        ```
+
+    Invalid:
+
+        ```python
+        delay[2 * a * a] $0;
+         ```
+
+- Stretch expressions can only include a single stretch variable.
+    Invalid:
+
+        ```python
+        stretch a;stretch b;
+        delay[ a + b ] $0;delay[ a * b ] $0;
```

Comment:
```python
a = circuit.add_stretch("a")
b = circuit.add_stretch("b")
circuit.delay(expr.add(a, b), q0)
circuit.delay(expr.mul(a, b), q0)
```

Side question also for @jakelishman, `expr.mul(a, b)` fails in Qiskit with `cannot multiply two durations`... why can't I multiple 2 durations ?

Reply:
The intent is to do a*b in dt, not "dt squared" - e.g. use `b` as a scalar without a unit. But I see your point - it'd need to somehow cast `b` to a scalar value first.